File: Laba2/4.swift
let nums = readLine()?.components(separatedBy: ", ") ?? []
var out = 0
for i in nums {
    if let number = Int(i) {
        let length = String(abs(number)).count
        if length % 2 != 0 {
            out += 1
        }
    }
}
print(out)

File: Laba2/4.rs
use std::io;

fn main() {
    println!("Введите числа через пробел: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input);
    let nums: Vec<i32> = input.split_whitespace().filter_map(|s| s.parse().ok()).collect(); // Как map в питоне, собирает в вектор
    let mut count = 0;
    for &num in &nums {
        let num_str = num.abs().to_string();
        if num_str.len() % 2 != 0 {
            count += 1;
        }
    }
    println!("{}", count);
}
